### PR TITLE
chore: configured eslint, prettier, and git hooks

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Had a chat with @ascorbic this morning and the linter rules that were previously committed were causing too much pain even though they're the ones we use in most projects, so we're going with the recommended eslint and @typescrint eslint rules along with the prettier plugin.

Scripts for formatting have been simplified as well relying on external configs and ignore files for eslint and prettier.

The deploy preview failing can be safely ignored. It's not related to these changes and is something I need to sort out after.

## Related Tickets & Documents

- Related Issue #21
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes_

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we
      can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or
      something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 📖. This ensures your code follows our style
      guide and passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

Hamsters playing instruments

### CODE REVIEW

**Comment by nickytonline** [comment]
The `deploy/netlify — Deploy Preview failed.` can be ignored. I have to sort that out but it's not related to the PR.

File: packages/remix-edge-adapter/src/index.ts [hunk 1]
```diff
@@ -1 +1 @@
-console.log('nothing for now')
+console.log('nothing for now');
```
Comment (nickytonline) [inline packages/remix-edge-adapter/src/index.ts:1]: prettier in action here and in files below

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -4,32 +4,35 @@ about: Create a report to help us improve
 title: ''
 labels: ''
 assignees: ''
----
 
-**Describe the bug** A clear and concise description of what the bug is.
```
Comment (nickytonline) [inline .github/ISSUE_TEMPLATE/bug_report.md:9]: This and other markdown files changed because of prettier.

File: package.json [hunk 3]
```diff
@@ -37,15 +30,19 @@
...
+    "@typescript-eslint/parser": "^5.50.0",
+    "commitlint": "^17.4.2",
+    "eslint": "^8.33.0",
+    "eslint-prettier-config": "^1.0.1",
+    "husky": "^8.0.3",
+    "lint-staged": "^13.1.0",
     "prettier": "^2.8.3",
-    "typescript": "^4.9.4",
+    "run-s": "^0.0.0",
+    "typescript": "^4.9.5",
```
Comment (nickytonline) [inline package.json:42]: a TS update crept in here.

File: .prettierrc.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
-  ...require('@netlify/eslint-config-node/.prettierrc.json'),
-  endOfLine: 'auto',
-}
+  useTabs: false,
+  singleQuote: true,
+  trailingComma: 'all',
```
Comment (ascorbic) [inline .prettierrc.js:4]: It should be `semi: false` too, to match the rest of our code

File: packages/remix-edge-adapter/src/index.ts (hunk 1 above)
Comment (ascorbic) [reply packages/remix-edge-adapter/src/index.ts:1]: We should keep semi false

File: .prettierrc.js (hunk 4 above)
Comment (ascorbic) [reply .prettierrc.js:4]: I think just copy https://github.com/netlify/eslint-config-node/blob/main/.prettierrc.json for now

File: package.json [hunk 5]
```diff
@@ -4,22 +4,15 @@
...
-    "eslint": "--cache --format=codeframe --max-warnings=0 \"{packages,src,scripts,tests,.github}/**/*.{ts,js,md,html}\" \"*.{ts,js,md,html}\" \".*.{ts,js,md,html}\"",
-    "prettier": "--loglevel=warn \"{packages,src,scripts,tests,.github}/**/*.{ts,js,md,yml,json,html}\" \"*.{ts,js,yml,json,html}\" \".*.{ts,js,yml,json,html}\" \"!package-lock.json\""
+    "coverage": "vitest run --coverage",
+    "prepare": "husky install",
+    "format": "run-s format:fix:*",
+    "format:ci": "run-s format:check:*",
+    "format:check:lint": "eslint --cache .",
+    "format:fix:lint": "eslint --fix --cache .",
+    "format:check:prettier": "prettier --cache .",
+    "format:fix:prettier": "prettier --write --cache ."
```
Comment (ascorbic) [inline package.json:15]: So clean!

File: .husky/commit-msg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
Comment (nickytonline) [inline .husky/commit-msg:1]: This is a git hook that runs to lint commit messages.

File: .husky/pre-commit [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env sh
```
Comment (nickytonline) [inline .husky/pre-commit:1]: This is the pre-commit hook that runs linting and formatting. It will error if there is an error that can't be autofixed.

**Review by sarahetter — APPROVED** [review]

**Pull request merged**